snap/backend/src/services/symbolClient.js

```diff
@@ -0,0 +1,83 @@
+import fetchUtils from '../utils/fetchUtils.js';
+import { utils } from 'symbol-sdk';
+import { Address } from 'symbol-sdk/symbol';
+
+const symbolClient = {
+	create(nodeUrl) {
+		return {
+			/**
+			 * Fetch native mosaic from network properties.
+			 * @returns {Promise<string>} the network currency mosaic id
+			 */
+			fetchNetworkCurrencyMosaicId: async () => {
+				try {
+					const { chain } = await fetchUtils.fetchData(`${nodeUrl}/network/properties`);
+					return chain.currencyMosaicId.slice(2).replace(/'/g, '');
+				} catch (error) {
+					throw new Error(`Failed to fetch network properties info: ${error.message}`);
+				}
+			},
+			/**
+			 * Fetch mosaics info from mosaic ids.
+			 * @param {Array<string>} mosaicIds - The mosaic ids to fetch info.
+			 * @returns {Promise<object<string, MosaicsInfo>>} The mosaics info.
+			 */
+			fetchMosaicsInfo: async mosaicIds => {
+				if (!mosaicIds.length)
+					return {};
+
+				try {
+					const mosaics = await fetchUtils.fetchData(`${nodeUrl}/mosaics`, 'POST', { mosaicIds });
+
+					return mosaics.reduce((mosaicObj, { mosaic }) => {
```

Comment:
Why reduce and not map?

Reply:
Because the goal is to create an Object, which **id** as `key`, and **{ divisibility }** as `value`.

If using `map` is will be an array of objects. 

example:

```json
{
  "0x123": {
    "divisibility": 6
  }
}
```